TDesignResolver结合TMap时会引入tdesign-vue/es/map/style，导致报错，能否忽略TMap?

### Describe the bug

``TDesignResolver结合TMap时会引入tdesign-vue/es/map/style，导致报错，能否忽略TMap?
vue.config.js
```
const AutoImport = require('unplugin-auto-import/webpack');
const Components = require('unplugin-vue-components/webpack');
const { TDesignResolver } = require('unplugin-vue-components/resolvers');
module.exports ={
configureWebpack: {
 plugins: [
      AutoImport({
        resolvers: [TDesignResolver()],
      }),
      Components({
        resolvers: [TDesignResolver()],
      }),
    ]
}
}
```
HomeView.vue

```
<template>
  <div class="home">
    <div id="container" />
  </div>
</template>

<script>
window.map = null;

export default {
  name: 'HomeView',
  mounted() {
    this.initMap();
  },
  methods: {
    initMap() {
      // 初始化地图
      window.map = new TMap.Map('container', {
        center: new TMap.LatLng(40.040452, 116.273486), // 地图显示中心点
        zoom: 20,	// 缩放级别
        rotation: 333.6,
        baseMap: {  // 设置卫星地图
          type: 'satellite',
          features: ['base'],
        },
      });
    },
  },
};
</script>
```

报错
```
 ERROR  Failed to compile with 1 error                                                                                                                                                    11:49:21

 error  in ./src/views/HomeView.vue?vue&type=script&lang=js&

Module not found: Error: Can't resolve 'tdesign-vue/es/map/style' in '/Users/brucezmlan/dev/projects/8p_lz/workspace/traffic-map-integration-platform-fe/src/views'

ERROR in ./src/views/HomeView.vue?vue&type=script&lang=js& (./node_modules/unplugin/dist/webpack/loaders/transform.js?unpluginName=unplugin-vue-components!./node_modules/unplugin/dist/webpack/loaders/transform.js?unpluginName=unplugin-auto-import!./node_modules/babel-loader/lib/index.js??clonedRuleSet-40.use[0]!./node_modules/@vue/vue-loader-v15/lib/index.js??vue-loader-options!./src/views/HomeView.vue?vue&type=script&lang=js&) 2:0-34
Module not found: Error: Can't resolve 'tdesign-vue/es/map/style' in '/Users/brucezmlan/dev/projects/8p_lz/workspace/traffic-map-integration-platform-fe/src/views'
 @ ./src/views/HomeView.vue?vue&type=script&lang=js& 1:38-435 1:451-454 1:456-850 1:456-850
 @ ./src/views/HomeView.vue 2:0-60 3:0-55 3:0-55 10:2-8
 @ ./src/router/index.js 3:0-45 8:13-21
 @ ./src/main.js 11:0-30 16:2-8

webpack compiled with 1 error
```



### Reproduction

Minimal

### System Info

```Shell
System:
    OS: macOS 12.6.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 2.16 GB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.15.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.5.5 - /usr/local/bin/npm
    Watchman: 2022.09.19.00 - /usr/local/bin/watchman
  Browsers:
    Chrome: 107.0.5304.110
    Firefox: 102.0
    Safari: 16.1
```


### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/src/core/resolvers/tdesign.ts
+++ b/src/core/resolvers/tdesign.ts
@@ -25,6 +25,12 @@ export interface TDesignResolverOptions {
    * @default false
    */
   esm?: boolean
+
+  /**
+   * exclude component name, if match do not resolve the name
+   *
+   */
+  exclude?: string | RegExp | (string | RegExp)[]
 }
 
 function getSideEffects(importName: string, options: TDesignResolverOptions): SideEffectsInfo | undefined {
@@ -91,8 +97,11 @@ export function TDesignResolver(options: TDesignResolverOptions = {}): Component
   return {
     type: 'component',
     resolve: (name: string) => {
-      const { library = 'vue' } = options
+      const { library = 'vue', exclude } = options
       const importFrom = options.esm ? '/esm' : ''
+
+      if (options.exclude && isExclude(name, exclude))
+        return
 
       if (options.resolveIcons && name.match(/[a-z]Icon$/)) {
         return {
@@ -112,4 +121,20 @@ export function TDesignResolver(options: TDesignResolverOptions = {}): Component
       }
     },
   }
+}
+
+function isExclude(name: string, exclude: string | RegExp | (string | RegExp)[] | undefined): boolean {
+  if (typeof exclude === 'string')
+    return name === exclude
+
+  if (exclude instanceof RegExp)
+    return !!name.match(exclude)
+
+  if (Array.isArray(exclude)) {
+    for (const item of exclude) {
+      if (name === item || name.match(item))
+        return true
+    }
+  }
+  return false
 }

--- a/test/resolvers/tdesign.test.ts
+++ b/test/resolvers/tdesign.test.ts
@@ -1,0 +1,22 @@
+import { describe, expect, it } from 'vitest'
+import { TDesignResolver } from '../../src/resolvers'
+
+import type { ComponentResolveResult, ComponentResolverObject } from '../../src'
+
+describe('TDesignResolver', () => {
+  it('name matching string rule should not be resolved', async () => {
+    const resolver = TDesignResolver({ exclude: 'TString' }) as ComponentResolverObject
+    expect(resolver.resolve('TString')).toBeFalsy()
+  })
+
+  it('name matching RegExp rule should not be resolved', async () => {
+    const resolver = TDesignResolver({ exclude: /^TDoc[A-Z]/ }) as ComponentResolverObject
+    expect(resolver.resolve('TDocRegExp')).toBeFalsy()
+  })
+
+  it('name matching Array<string | RegExp> rule should not be resolved', async () => {
+    const resolver = TDesignResolver({ exclude: ['TString', /^TDoc[A-Z]/] }) as ComponentResolverObject
+    expect(resolver.resolve('TString')).toBeFalsy()
+    expect(resolver.resolve('TDocRegExp')).toBeFalsy()
+  })
+})

